examples/bzlmod/entry_points/BUILD.bazel

```diff
@@ -0,0 +1,95 @@
+load("@bazel_skylib//rules:run_binary.bzl", "run_binary")
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
+load("@python_versions//3.9:defs.bzl", py_console_script_binary_3_9 = "py_console_script_binary")
+load("@rules_python//python:defs.bzl", "py_test")
+load("@rules_python//python/entry_points:py_console_script_binary.bzl", "py_console_script_binary")
+
+# This is how you can define a `pylint` entrypoint which uses the default python version.
+py_console_script_binary(
+    name = "pylint",
+    pkg = "@pip//pylint",
+)
+
+# And a test that we can correctly run `pylint --version`
+py_test(
+    name = "pylint_test",
+    srcs = ["pylint_test.py"],
+    data = [":pylint"],
+    env = {
+        "ENTRY_POINT": "$(rlocationpath :pylint)",
+    },
+    deps = ["@rules_python//python/runfiles"],
+)
+
+# We can also specify extra dependencies for the binary, which is useful for
+# tools like flake8, pylint, pytest, which have plugin discovery methods.
+py_console_script_binary(
+    name = "pylint_with_deps",
+    pkg = "@pip//pylint",
+    # Because `pylint` has multiple console_scripts available, we have to
+    # specify which we want if the name of the target name 'pylint_with_deps'
+    # cannot be used to guess the entry_point script.
+    script = "pylint",
+    deps = [
+        # One can add extra dependencies to the entry point.
+        "@pip//pylint_print",
+    ],
+)
+
+# Here we create a file that has some pylint errors detected only with the
```

Comment:
Thanks, whilst updating the docs I wanted to have a rule that would allow me to take code snippets from the `examples/entry_points/BUILD.bazel` and your suggestion of moving all of the test code to a different `BUILD.bazel` makes a lot of sense at least in a sense that the comments directed towards maintainers will be next to the tests and the comments directed to readers of the examples will be next to the target definitions.

Moving them to `//entry_points/tests` along with the test files.